### ISSUE TITLE
Add complete Row Pattern Recognition grammar to Snowflake

### ISSUE THREAD

**Issue by cbows** [sqlfluff/sqlfluff#2783]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

In December 2016, the International Organization for Standardization (ISO) released a new version of the SQL standard which includes Row Pattern Recognition in SQL (ISO/IEC 9075:2016 or ANSI X3.135)

Snowflake supports this feature (see [Snowflake documentation](https://docs.snowflake.com/en/sql-reference/constructs/match_recognize.html)) as well as some other ANSI compliant SQL engines (e.g. [Trino](https://trino.io/docs/current/sql/match-recognize.html) (f.k.a. PrestoSQL)).

### Use case

It would be nice if SQLFluff supports this syntax.

### Dialect

ANSI
Snowflake

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jpy-git** [comment]
I've already got a PR (#2781) open atm to add the `MATCH_RECOGNIZE` grammar to Snowflake.

The key aim of that PR is to get `MATCH_RECOGNIZE` parsing. The actual `Pattern` grammar is fairly complex itself so for the initial PR I've limited it to symbols and quantifiers (which should cover a lot of the basic use cases) and then we can follow up afterwards to implement operators and complete the `Pattern` grammar.

Should be fairly simple to extrapolate out to other grammars afterwards.
